independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java

```diff
@@ -317,48 +325,135 @@ private static String configuredBuildDir(LocalProject project, Function<Build, S
-        module.addMainSources(new DefaultProcessedSources(getSourcesSourcesDir().toFile(), getClassesDir().toFile()));
-        addMainResources(module);
-        module.addTestSources(new DefaultProcessedSources(getTestSourcesSourcesDir().toFile(), getTestClassesDir().toFile()));
-        addTestResources(module);
+
+        final Build build = (modelBuildingResult == null ? getRawModel() : modelBuildingResult.getEffectiveModel()).getBuild();
+        if (build != null && !build.getPlugins().isEmpty()) {
+            for (Plugin plugin : build.getPlugins()) {
+                if (!plugin.getArtifactId().equals("maven-jar-plugin")) {
+                    continue;
+                }
+                for (PluginExecution e : plugin.getExecutions()) {
+                    DefaultArtifactSources src = null;
+                    if (e.getGoals().contains(GACTV.TYPE_JAR)) {
+                        src = processJarPluginExecutionConfig(e, false);
+                    } else if (e.getGoals().contains("test-jar")) {
+                        src = processJarPluginExecutionConfig(e, true);
+                    }
+                    if (src != null) {
+                        module.addArtifactSources(src);
+                    }
+                }
+            }
+        }
+
+        if (module.getMainSources() == null) {
+            final DefaultArtifactSources src = new DefaultArtifactSources(DefaultWorkspaceModule.MAIN);
+            src.addSources(new DefaultSourceDir(getSourcesSourcesDir().toFile(), getClassesDir().toFile()));
+            addMainResources(src, null);
+            module.addArtifactSources(src);
+        }
+        if (module.getTestSources() == null) {
+            final DefaultArtifactSources src = new DefaultArtifactSources(DefaultWorkspaceModule.TEST);
+            src.addSources(new DefaultSourceDir(getTestSourcesSourcesDir().toFile(), getTestClassesDir().toFile()));
+            addTestResources(src, null);
+            module.addArtifactSources(src);
+        }
         module.setBuildFiles(PathList.of(getRawModel().getPomFile().toPath()));
-        return module;
+
+        return this.module = module;
+    }
+
+    private DefaultArtifactSources processJarPluginExecutionConfig(PluginExecution e, boolean test) {
+        final Object config = e.getConfiguration();
+        if (config == null || !(config instanceof Xpp3Dom)) {
+            return null;
+        }
+        Xpp3Dom dom = (Xpp3Dom) config;
+        PathFilter filter = null;
+        final Xpp3Dom includes = dom.getChild("includes");
+        if (includes != null) {
+            final Xpp3Dom[] includeElements = includes.getChildren();
+            final List<String> list = new ArrayList<>(includeElements.length);
+            for (Xpp3Dom include : includeElements) {
+                list.add(include.getValue());
+            }
+            filter = PathFilter.forIncludes(list);
+        } else {
```

Comment:
Are you sure about ignoring excludes when there are includes?
E.g.: https://stackoverflow.com/a/2450803/9529981

Reply:
No, I'll fix it.

Reply:
Btw, this is a working link to that Scanner class: https://codehaus-plexus.github.io/plexus-utils/apidocs/org/codehaus/plexus/util/DirectoryScanner.html

Reply:
Thanks, we'll need to add proper tests for this.